get_web_content.pl

```diff
@@ -76,6 +77,18 @@
         my $img_file = $image->{file};
         $image->{outfile} = "$TaskData/$img_file";
         my $got_image = 0;
+
+        if (exists $web->{add_date}) { # Ugh, custom code for magnetogram
+            my $src = $image->{filter}{src};
+            my $year = strftime "%Y", gmtime;
+            my $mon = strftime "%b", gmtime;
+            my $day = strftime "%e", gmtime;
+            $mon = lc($mon);
+            $day =~ s/^\s+//;
+            my $str = "$year/$mon$day\_$year\_mag";
+            $image->{filter}{src} = $src ~=  s/mag/$str/r;
```

Comment:
Is this working around the issue that the "latest" magnetograms from solarham don't have the annotations that the last-dated ones have?

Reply:
> Is this working around the issue that the "latest" magnetograms from solarham don't have the annotations that the last-dated ones have?

Yes. But overall I suspect that this code won't work well in production because it assumes that the image for a particular date is available upstream at exactly midnight GMT of that date. Since this Replan Central code is running every few minutes we are likely to hit this case every day.

One idea is to scrape the image name from `view-source:https://solarham.com/magnetogram.htm`. It won't be be bulletproof but that file has this:
```
<TABLE cellspacing="0" cellpadding="5" border="0"><TR><TD width="800" valign="top"><center>
 <a href="pictures/2024/jun7_2024_mag.jpg">
 <img src="pictures/2024/jun7_2024_mag.jpg" height="760" width="760"></a><p>
</TD>
```
So you can get the HTML for `https://solarham.com/magnetogram.htm` and grep for `pictures/20\d\d/[a-z]+\d+_20\d\d_mag.jpg` and find the latest image name. Since the upstream file name is unique by day, it would be polite to check if we already have it and don't grab it again every 5 minutes.

Reply:
Not sure if this solution would work for the image suggested below by Scott, the available images are listed at https://www.solen.info/solar/images/ and the HTML where we could find out the latest name does not seem to be available. Would it be useful to add a new field to the config file, e.g., "fetch_every_n_hours" to do something like <code>if (defined $image->{fetch_every_n_hours}) {<check the age of the solar_flare_monitor image we have> <proceed with fetching the image if the age is older than the defined age>}</code>, similarly to the logic behind warning based on warn_age_hours?

Reply:
Regarding "it would be polite to check if we already have it": arc stores the solar monitor file as solar_flare_monitor.png, so we loose the date info contained in the file name. It makes more sense to me to decide if a new file needs to be grabbed based on the image age since last modified.

Reply:
Additionally, as of 06/07/2024 7pm, the most recent image on https://www.solen.info/solar/images/ is still AR_CH_20240606.png, so it may be that it updates with a 24h delay with respect to the current date.